sim.c

```diff
@@ -50,7 +50,7 @@ static Usz index_of(Glyph c) {
 #else
 // Reference implementation
 static Usz index_of(Glyph c) {
-  if (c == '.')
+  if (c == '.' || c == '*')
```

Comment:
This is unnecessary. The comparison against `'.'` is for the common case early-out. It's listed first because it's the most common case in most .orca files. `'*'` is the rare case and doesn't need to have a comparison added for it explicitly. It will fall through to the end and return 0.

Reply:
It's considered more polite and easier to deal with if commits designed for pull requests are separated logically instead of unrelated changes being lumped together.

Reply:
ok, but as that implementation is not used and it is only a reference, i thought we could be explicit about `*` being treated as 0. Should i make another commit to revert that? 

(sorry for the multiple things together, as there were many minor things i thought it would have made the review process faster)